cmd/tempo-query/tempo/plugin.go

```diff
@@ -316,19 +347,45 @@ func (b *Backend) FindTraces(req *storage_v1.FindTracesRequest, stream storage_v
 	}
 
 	span.AddEvent(fmt.Sprintf("Found %d trace IDs", len(resp.TraceIDs)))
+	b.logger.Info("FindTraces: fetching traces", zap.Int("traceids", len(resp.TraceIDs)))
+
+	numWorkers := b.findTracesConcurrentRequests
+	jobs := make(chan job, len(resp.TraceIDs))
+	results := make(chan jobResult, len(resp.TraceIDs))
+	var workersDone sync.WaitGroup
+	// Start workers
+	for w := 0; w < numWorkers; w++ {
+		workersDone.Add(1)
+		go func() { defer workersDone.Done(); worker(b, jobs, results) }()
+	}
 
 	// for every traceID, get the full trace
 	var jaegerTraces []*jaeger.Trace
 	for _, traceID := range resp.TraceIDs {
-		trace, err := b.getTrace(ctx, traceID)
-		if err != nil {
-			// TODO this seems to be an internal inconsistency error, ignore so we can still show the rest
-			span.AddEvent(fmt.Sprintf("could not get trace for traceID %v", traceID))
-			span.RecordError(err)
-			continue
+		jobs <- job{
+			ctx:     ctx,
+			traceID: traceID,
 		}
-
-		jaegerTraces = append(jaegerTraces, trace)
+	}
+	close(jobs)
+	workersDone.Wait()
+
+	var failedTraces []jobResult
+	// Collecting results
+	for i := 0; i < len(resp.TraceIDs); i++ {
+		result := <-results
+		//// TODO this seems to be an internal inconsistency error, ignore so we can still show the rest
+		span.AddEvent(fmt.Sprintf("could not get trace for traceID %v", result.traceID))
```

Comment:
Should these two statements be moved within err block below?

Reply:
Fixed